File: Controllers/Api/TripsController.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using AutoMapper;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Logging;
using MyTravel.Models;
using MyTravel.ViewModels;

namespace MyTravel
{
    [Route("api/trips")]
    [Authorize]
    public class TripsController : Controller
    {
        private ITravelRepository _repository;
        private ILogger<TripsController> _logger;

        public TripsController(ITravelRepository repository, ILogger<TripsController> logger)
        {
            _repository = repository;
            _logger = logger;
        }

        [HttpGet]
        public IActionResult Get()
        {
            try
            {
                var trips = _repository.GetAllTripsWithStops(User.Identity.Name);
                return Ok(Mapper.Map<IEnumerable<TripViewModel>>(trips));
            }
            catch (Exception ex)
            {
                _logger.LogError($"Failed to get all trips: {ex}");
            }
            return BadRequest($"Error Occurs when get all Trips");
        }

        [HttpPost]
        public async Task<IActionResult> Post([FromBody] TripViewModel model)
        {
            if (ModelState.IsValid)
            {
                var newTrip = Mapper.Map<Trip>(model);
                newTrip.UserName = User.Identity.Name;
                _repository.AddTrip(newTrip);

                if (await _repository.SaveChangesAsync())
                {
                    var encodedName = System.Net.WebUtility.UrlEncode(newTrip.Name);
                    return Created($"api/trips/{encodedName}", Mapper.Map<TripViewModel>(newTrip));
                }
                else
                    return BadRequest("Failed to save changes to the database");
            }
            return BadRequest(ModelState);
        }
    }
}

File: wwwroot/js/tripsController.js
// tripsController.js
(function () {
    'use strict';

    angular.module("app-trips")
        .controller("tripsController", tripsController);
    tripsController.$inject = ['$http'];
    function tripsController($http) {
        var vm = this;
        vm.trips = [];

        vm.newTrip = {};

        vm.errorMessage = "";
        vm.isBusy = true;

        $http.get("/api/trips").then(
            function (response) {
                // success
                angular.copy(response.data, vm.trips);
            },
            function (error) {
                // failure
                vm.errorMessage = "Failed to Load data: " + error;
            }
        ).finally(function () {
            // vm.isBusy = false;
        });

        vm.addTrip = function () {
            vm.isBusy = true;

            $http.post("/api/trips", vm.newTrip).then(
                function (response) {
                    vm.trips.push(response.data);
                    vm.newTrip = {};
                },
                function (error) {
                    vm.errorMessage = "Failed to Load data: " + error;
                }
            ).finally(function () {
                vm.isBusy = false;
            });
        }
    }
})();